### PR TITLE
Zenodo CSV: Figure 5 and part of S6

### PR DESCRIPTION
**Pull request by sjspielman**
Towards #1692
**⚠️  Stacked on #1707**

This PR adds CSVs for all [figure 5](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/figures/pngs/figure5.png) panels, and the non-UMAP [figure S6 panels](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/figures/pngs/figureS6.png), because they are in one of the figure 5 scripts.

#### 5A & 5B: `figures/scripts/fig5-panels-gsva-umap.R`

- 5A is the UMAP. This was pretty straight forward, and you might like some of the piping changes. One change I made was removing the line `select(X1, X2, broad_histology_display)`, since this ends up removing sample ID information which we'd like to keep in the export, and it doesn't hurt to be there during plotting.
- 5B is the GSVA heatmap. I decided to (but feel 50/50 about it) transpose the data for export so that the biospecimen ID would be a column itself instead of column names. The exported (wide) CSV therefore has pathways as column names. 

#### 5D: `figures/scripts/fig5-forest-plot.R`

This is another forest plot. You'll be happy to see the code is about the same for all forest plots in the paper! I exported the non-spread data frame.

#### 5C, 5E, S6E, S6F: `figures/scripts/fig5_figS6-immune-deconv-panels.R`

These are all pretty straightforward plots, nothing too interesting about these during export. In the end, the code was all roughly the same for each of the 4 exports, so I suppose I could have written a little wrapper function, but I did not...

### CODE REVIEW

**Review by jashapiro — COMMENTED** [review]
UMAP plot (and other) outputs including hex values seems a bit excessive, but I'm not sure it is worth removing.

I also made a small suggestion about removing redundant info, which might apply to earlier output that I missed... Otherwise all looks good.

File: figures/scripts/fig5_figS6-immune-deconv-panels.R [hunk 1]
```diff
@@ -396,3 +404,54 @@ ggsave(cd8_cd4_ratio_pdf,
...
        width = 3.5, height = 4, useDingbats = FALSE)
 
+
+
+# Export CSVs for Zenodo upload
+
+# Panel 5C
+quantiseq_cg %>%
+  # reorder columns so the ID is first
+  dplyr::select(Kids_First_Biospecimen_ID = sample, everything()) %>%
```
[the remark this answers is not in the document]
Comment (sjspielman) [reply figures/scripts/fig5_figS6-immune-deconv-panels.R:414]: Yeah, lots of that going around in this `dplyr` version...

File: figures/scripts/fig5_figS6-immune-deconv-panels.R [hunk 2]
```diff
@@ -396,3 +404,54 @@ ggsave(cd8_cd4_ratio_pdf,
...
+
+
+# Panel 5E
+CD274_cd8_mb %>%
+  # reorder columns so the ID is first
+  dplyr::select(Kids_First_Biospecimen_ID = sample, everything()) %>%
+  # arrange on sample
+  dplyr::arrange(Kids_First_Biospecimen_ID) %>% 
+  # remove \n from molecular_subtype so that CSV is properly formatted
+  dplyr::mutate(molecular_subtype = stringr::str_replace(molecular_subtype, "\n", " ")) %>%
```
[the remark this answers is not in the document]
Comment (sjspielman) [reply figures/scripts/fig5_figS6-immune-deconv-panels.R]: Noting I'm doing this swap throughout where relevant

**Review by jashapiro — APPROVED** [review]
LGTM...

Re 
> In general, I think it needs to be there when we've used* scale_color/fill_identity(), but otherwise it's probably just noise (although harmless!).

I was thinking that our color choices don't really need to be part of the data export, as if people want to reproduce using our precise code, they are probably using other inputs, not the zenodo export, but as I said, it's not really worth the effort to remove either.

**Pull request merged**